src/helpers/routes.js

```diff
@@ -77,7 +77,6 @@ exports.makeRoutes = (baseRoutes, {
 
       // Generate localized children routes if any
       if (route.children) {
-        delete localizedRoute.name
```

Comment:
Actually I found out what issue removing route name was solving. In the case of directory structure like:
```
pages:
 - posts/index.vue
 - posts.vue
```
Nuxt will generate routes:
```
{
    path: '/posts',
    component: _f3e2cabc,
    children: [{
      path: '',
      component: _50b11fa5,
      name: 'posts'
    }]
  }
```

So the parent route doesn't have a name in this case, and if it had, VueRouter would trigger warning that child route is not accessible using `:to({ name: 'posts' })`.

So I need to look more into it. I guess the name should only be removed when there will be no child routes that would "shadow" parent's name.

Reply:
Addendum: This is an issue *when combined with* the fact that we are always setting name for extra routes added for locales. Then we can end up with multiple identical names that shadow each other.

Reply:
I believe I have it figured out now.